### ISSUE TITLE
[HELP NEEDED : INSTALLATION]

### ISSUE THREAD

**Issue by arifsnoor** [BlackArch/blackarch#3394]
##How to format partitions in FAT32 during installation?

I was trying to install Blackarch in GPT mode. When the format option came I cant format the EFI Boot partition as FAT32. It selects ext4 as default. I tried inputting 'fat -F32' which shows error formatting. I tried virtual box and single boot in hard drive. Both failed. I can't even find anything online. Please help me format the boot partition. Thank you.

![Screenshot 2022-04-25 114819](https://user-images.githubusercontent.com/68831776/165029788-16bed6e8-f4a9-44ed-97f6-27e3519aed92.png)
![Screenshot 2022-04-25 115022](https://user-images.githubusercontent.com/68831776/165029807-97f18b49-6976-4fff-a676-fdd80897a3f2.png)
![Screenshot 2022-04-25 115300](https://user-images.githubusercontent.com/68831776/165029820-0a27e114-21ec-4dd7-b060-def057b699b5.png)

**Comment by anunna** [comment]
it's called `vfat`

**Comment by arifsnoor** [comment]
> it's called `vfat`

Still showing error. I assume it only takes ext file system.

**Comment by anunna** [comment]
Yes, if you booted UEFI and are asked to point the mount points, you will only need to enter file systems for everything other than EFI. If you are asked to enter the file system type for boot, then you booted MBR and will not work.

`vfat` is valid and quicker to type - [here](https://wiki.archlinux.org/title/FAT#File_system_creation)

**Comment by dcunited001** [comment]
to determine whether the system is efi, =/sbin/blackarch-install= runs `$(efivar --list)`, but this command is missing for me. i'm using the full ISO.

a simple workaround: the variable `BOOT_MODE` that is set by this command needs to be changed in the script. it is set to empty string upon init. edit the script and set `BOOT_MODE=uefi`

it seems to be working for me, but if the install fails, it's possible that the source for =blackarch-install= will be replaced by pacman
